client/components/PubEdge/PubEdge.tsx

```diff
@@ -93,7 +95,14 @@ const PubEdge = (props: PubEdgeProps) => {
 			})}
 			titleElement={maybeLink(title)}
 			bylineElement={
-				contributors && contributors.length > 0 && <Byline contributors={contributors} />
+				contributors &&
+				contributors.length > 0 &&
+				(externalPublication ? (
+					<Byline contributors={contributors} />
+				) : (
+					// I know there's a better way to do this with a typeguard.
+					<WithinCommunityByline contributors={contributors as AttributionWithUser[]} />
+				))
```

Comment:
I think the "ideal" way to solve this issue would be in the type definition of PubEdge. Basically PubEdge would have two variants:
```ts
type BasePubEdge = { /* shared attributes */ }
type ExternalPubEdge = BasePubEdge & { contributors: string[], externalPublication: ExternalPublication }
type InternalPubEdge = BasePubEdge & { contributors: AttributionWithUser[], pub: Pub }
type PubEdge = ExternalPublication | InternalPublication
```

Then, just checking `"externalPublication" in pubEdge` would automatically narrow the type of `pubEdge` from `ExternalPubEdge | InternalPubEdge` to `ExternalPubEdge` or `InternalPubEdge`:

```ts
"externalPublication" in pubEdge ?
  // narrowed to `ExternalPubEdge`
  ? <Byline contributors={pubEdge.contributors} />
  // narrowed to `InternalPubEdge`
  : <WithinCommunityByline contributors={pubEdge.contributors} />
```